Makefile

```diff
@@ -67,9 +78,14 @@ $(tool_dir).tar.xz: \
 		target/release/README.md
 	mkdir -p $(tool_dir)
 	cd target/release && cp --reflink=auto -t ../../$(tool_dir) $(files)
-	strip luxtorpeda/luxtorpeda
+
+$(tool_dir).tar.xz: $(tool_dir)
+	$(STRIP) luxtorpeda/luxtorpeda
```

Comment:
Hmm, there was a problem with my Makefile, that existed before your changes, but had no serious consequences (phony dependency in tarball target resulted in unncecessary re-creation of a tarball on each re-creation).

Now, with splitting tarball creation into 2 separate targets and invoking a strip only in one, this pre-existing problem got elevated into a small buildsystem bug:

    $ make clean
    $ make luxtorpeda.tar.xz   # binary in `$tool_dir` is stripped
    $ make luxtorpeda.tar.xz   # `$tool_dir` got re-created due to phony target in dependency, but strip is not invoked, because tarball already exists

To make the problem non-cosequential again, please move stripping to target `$(tool_dir)`. It means binaries used for `install` will be stripped as well, so that would be expected, I guess.

If someone will ever need unstripped tarball/installation, it will still be possible with:

    $ make STRIP=true luxtorpeda.tar.xz

but `STRIP=true` can be confusing (because it will result in no stripping in fact). For this reason I suggest replacing variable `STRIP` with `STRIP_PROG` (unless there are some reasons for keeping the name `STRIP`?).

After strip will be moved, I'll merge your changes and deal with removing unnecessary phony dependency.

Reply:
Do you need stripping in the makefile at all? In general, the buildsystem would strip binaries by itself _after_ the program is built. And for user-installed case you provide unstripped binaries anyway.

Reply:
I am not sure why cargo does not strip binaries - haven't looked into details yet. AFAIR from a few days ago - release binary is already ~10x smaller than a debug one, and stripping additionally cuts the size in half.

The original reason for stripping binaries was to make tarballs small enough, so they could be uploaded to Discord for other people to test (the limit is 10MB, I think). Right now, tarballs are automatically re-created on every push to GitLab and served from static site, but only on `master`. I would rather keep the stripping as default.

`user-install` is for development and debugging purposes at the moment, I don't want to strip these binaries, as Rust generates very nice callstacks in case I'll manage to outfox the compiler ;) (happened once so far).

Reply:
I believe stripping shouldn't happen during build phase by default. It's because package building systems generally want the debug symbols to be put in some additional package, not just thrown away.

Reply:
I see your point. At the moment defaults are designed to speed up my work; I will add `user.mk` file support to make the defaults overridable without creating scripts or aliases - after that, I could switch `STRIP` to be `true` by default.